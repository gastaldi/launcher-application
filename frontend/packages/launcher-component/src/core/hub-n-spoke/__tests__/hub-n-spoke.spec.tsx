import * as React from 'react';
import { HubNSpoke } from '..';
import { cleanup, fireEvent, render } from 'react-testing-library';
import { mockItems } from './mock-items';

afterEach(cleanup);

describe('<HubNSpoke />', () => {
  it('renders the HubNSpoke correctly', () => {
    const comp = render(<HubNSpoke title="HubNSpoke" items={mockItems} />);
    expect(comp.asFragment()).toMatchSnapshot();
  });

  it('show the edition panel for the clicked hub', () => {
    const comp = render(<HubNSpoke title="HubNSpoke" items={mockItems} />);
    const editHub1Button = comp.getByLabelText('Open hub1 editor');
    fireEvent.click(editHub1Button);
    expect(comp.getByText('this is hub 1 edition form')).toBeDefined();
    expect(comp.asFragment()).toMatchSnapshot();
    fireEvent.click(comp.getByLabelText('Close hub1 editor'));
    expect(comp.getByText('this is hub 2 overview')).toBeDefined();
    expect(comp.asFragment()).toMatchSnapshot();
  });
});
