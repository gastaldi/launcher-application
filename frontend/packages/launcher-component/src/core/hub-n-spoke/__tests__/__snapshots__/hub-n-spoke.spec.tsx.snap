// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HubNSpoke /> renders the HubNSpoke correctly 1`] = `
<DocumentFragment>
  <div
    class="hubNSpoke"
  >
    <h1
      class="hub-and-spoke-title"
      data-pf-content="true"
    >
      HubNSpoke
    </h1>
    <div
      class="pf-l-grid pf-m-gutter hub-and-spoke-container"
    >
      <div
        class="pf-l-grid__item pf-m-6-col-on-sm pf-m-3-col-on-md hub-and-spoke-item"
      >
        <div
          class="hub-and-spoke-header"
        >
          <h1>
            Hub1
          </h1>
          <div
            class="hub-and-spoke-nav"
          >
            <button
              aria-label="Open hub1 editor"
              class="pf-c-button pf-m-plain"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 576 512"
                width="1em"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  transform=""
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="hub-and-spoke-body"
        >
          <p>
            this is hub 1 overview
          </p>
        </div>
      </div>
      <div
        class="pf-l-grid__item pf-m-6-col-on-sm pf-m-3-col-on-md hub-and-spoke-item"
      >
        <div
          class="hub-and-spoke-header"
        >
          <h1>
            Hub2
          </h1>
          <div
            class="hub-and-spoke-nav"
          >
            <button
              aria-label="Open hub2 editor"
              class="pf-c-button pf-m-plain"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 576 512"
                width="1em"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  transform=""
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="hub-and-spoke-body"
        >
          <p>
            this is hub 2 overview
          </p>
        </div>
      </div>
      <div
        class="pf-l-grid__item pf-m-12-col-on-sm pf-m-12-col-on-md hub-and-spoke-item"
      >
        <div
          class="hub-and-spoke-header"
        >
          <h1>
            Hub3
          </h1>
          <div
            class="hub-and-spoke-nav"
          >
            <button
              aria-label="Open hub3 editor"
              class="pf-c-button pf-m-plain"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 576 512"
                width="1em"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  transform=""
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="hub-and-spoke-body"
        >
          <p>
            this is hub 3 overview
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<HubNSpoke /> show the edition panel for the clicked hub 1`] = `
<DocumentFragment>
  <div
    class="hubNSpoke"
  >
    <h1
      class="hub-and-spoke-title"
      data-pf-content="true"
    >
      HubNSpoke
    </h1>
    <div
      class="pf-l-grid pf-m-gutter hub-and-spoke-container"
    >
      <div
        aria-label="Edit hub1"
        class="pf-l-grid__item pf-m-12-col hub-and-spoke-item hub-and-spoke-item-form"
      >
        <div
          class="hub-and-spoke-header"
        >
          <h1>
            Hub1
          </h1>
          <div
            class="hub-and-spoke-nav"
          >
            <button
              aria-label="Close hub1 editor"
              class="pf-c-button pf-m-plain"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
                  transform=""
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="hub-and-spoke-body"
        >
          <p>
            this is hub 1 edition form
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<HubNSpoke /> show the edition panel for the clicked hub 2`] = `
<DocumentFragment>
  <div
    class="hubNSpoke"
  >
    <h1
      class="hub-and-spoke-title"
      data-pf-content="true"
    >
      HubNSpoke
    </h1>
    <div
      class="pf-l-grid pf-m-gutter hub-and-spoke-container"
    >
      <div
        class="pf-l-grid__item pf-m-6-col-on-sm pf-m-3-col-on-md hub-and-spoke-item"
      >
        <div
          class="hub-and-spoke-header"
        >
          <h1>
            Hub1
          </h1>
          <div
            class="hub-and-spoke-nav"
          >
            <button
              aria-label="Open hub1 editor"
              class="pf-c-button pf-m-plain"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 576 512"
                width="1em"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  transform=""
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="hub-and-spoke-body"
        >
          <p>
            this is hub 1 overview
          </p>
        </div>
      </div>
      <div
        class="pf-l-grid__item pf-m-6-col-on-sm pf-m-3-col-on-md hub-and-spoke-item"
      >
        <div
          class="hub-and-spoke-header"
        >
          <h1>
            Hub2
          </h1>
          <div
            class="hub-and-spoke-nav"
          >
            <button
              aria-label="Open hub2 editor"
              class="pf-c-button pf-m-plain"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 576 512"
                width="1em"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  transform=""
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="hub-and-spoke-body"
        >
          <p>
            this is hub 2 overview
          </p>
        </div>
      </div>
      <div
        class="pf-l-grid__item pf-m-12-col-on-sm pf-m-12-col-on-md hub-and-spoke-item"
      >
        <div
          class="hub-and-spoke-header"
        >
          <h1>
            Hub3
          </h1>
          <div
            class="hub-and-spoke-nav"
          >
            <button
              aria-label="Open hub3 editor"
              class="pf-c-button pf-m-plain"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 576 512"
                width="1em"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  transform=""
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="hub-and-spoke-body"
        >
          <p>
            this is hub 3 overview
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
